Python/symtable.c

```diff
@@ -910,13 +975,23 @@ analyze_block(PySTEntryObject *ste, PyObject *bound, PyObject *free,
         assert(c && PySTEntry_Check(c));
         entry = (PySTEntryObject*)c;
 
+        PySTEntryObject *new_class_entry = NULL;
+        if (entry->ste_can_see_class_scope) {
+            if (ste->ste_type == ClassBlock) {
+                new_class_entry = ste;
+            }
+            else if (class_entry) {
+                new_class_entry = class_entry;
+            }
```

Comment:
What is the case where we have multiple nested scopes that can all see the same enclosing class scope? Do you have tests for such cases?

Reply:
It's when there are typevar bounds. `type A[T: int] = float`.